aws/resource_aws_waf_web_acl.go

```diff
@@ -219,3 +308,48 @@ func resourceAwsWafWebAclDelete(d *schema.ResourceData, meta interface{}) error
 	}
 	return nil
 }
+
+func expandLoggingConfiguration(l []interface{}) *waf.LoggingConfiguration {
+	if l == nil || len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+
+	lc := &waf.LoggingConfiguration{
+		LogDestinationConfigs: []*string{
+			aws.String(m["log_destination"].(string)),
+		},
+	}
+
+	rf := m["redacted_fields"].([]interface{})
+	rfm := rf[0].(map[string]interface{})
+	ftm := rfm["field_to_match"].(*schema.Set).List()
+
+	for _, f := range ftm {
+		lc.RedactedFields = append(lc.RedactedFields, expandFieldToMatch(f.(map[string]interface{})))
+	}
+	return lc
+}
+
+func flattenLoggingConfiguration(n *waf.LoggingConfiguration) []map[string]interface{} {
+	if n == nil {
+		return nil
+	}
+
+	m := make(map[string]interface{})
+	m["log_destination"] = *n.LogDestinationConfigs[0]
+
+	m["redacted_fields"] = flattenRedactedFields(n.RedactedFields)
+	return []map[string]interface{}{m}
+}
+
+func flattenRedactedFields(ts []*waf.FieldToMatch) []interface{} {
```

Comment:
Here is a working example of this function for configurations with and without `redacted_fields`:

```go
func flattenRedactedFields(fieldToMatches []*waf.FieldToMatch) []interface{} {
	if len(fieldToMatches) == 0 {
		return []interface{}{}
	}

	fieldToMatchResource := &schema.Resource{
		Schema: map[string]*schema.Schema{
			"data": {
				Type:     schema.TypeString,
				Optional: true,
			},
			"type": {
				Type:     schema.TypeString,
				Required: true,
			},
		},
	}
	l := make([]interface{}, len(fieldToMatches))

	for i, fieldToMatch := range fieldToMatches {
		l[i] = flattenFieldToMatch(fieldToMatch)[0]
	}

	m := map[string]interface{}{
		"field_to_match": schema.NewSet(schema.HashResource(fieldToMatchResource), l),
	}

	return []interface{}{m}
}
```